counterpartylib/lib/util.py

```diff
@@ -198,6 +207,89 @@ def get_asset_name (db, asset_id, block_index):
+    chunks = asset.split('.', 1)
+    if (len(chunks) == 2):
+        subasset_parent = chunks[0]
+        subasset_child = chunks[1]
+        subasset_longname = asset
+
+        # validate parent asset
+        validate_subasset_parent_name(subasset_parent)
+
+        # validate child asset
+        validate_subasset_longname(subasset_longname, subasset_child)
+
+    return (subasset_parent, subasset_longname)
+
+# throws exceptions for invalid subasset names
+def validate_subasset_longname(subasset_longname, subasset_child=None):
+    if subasset_child is None:
+        chunks = subasset_longname.split('.', 1)
+        if (len(chunks) == 2):
+            subasset_child = chunks[1]
+        else:
+            subasset_child = ''
+
+    if len(subasset_child) < 1:
+        raise exceptions.AssetNameError('subasset name too short')
+    if len(subasset_longname) > 250:
+        raise exceptions.AssetNameError('subasset name too long')
+
+    # can't start with period, can't have consecutive periods, can't contain anything not in SUBASSET_DIGITS
+    previous_digit = '.'
+    for c in subasset_child:
+        if c not in SUBASSET_DIGITS:
+            raise exceptions.AssetNameError('subasset name contains invalid character:', c)
+        if c == '.' and previous_digit == '.':
+            raise exceptions.AssetNameError('subasset name contains consecutive periods')
+        previous_digit = c
+    if previous_digit == '.':
+        raise exceptions.AssetNameError('subasset name ends with a period')
+
+    return True
+
+# throws exceptions for invalid subasset names
+def validate_subasset_parent_name(asset_name):
+    if asset_name == config.BTC:
+        raise exceptions.AssetNameError('parent asset cannot be {}'.format(config.BTC))
+    if asset_name == config.XCP:
+        raise exceptions.AssetNameError('parent asset cannot be {}'.format(config.XCP))
+    if len(asset_name) < 4:
+        raise exceptions.AssetNameError('parent asset name too short')
+    if len(asset_name) >= 13:
+        raise exceptions.AssetNameError('parent asset name too long')
+    if asset_name[0] == 'A':
+        raise exceptions.AssetNameError('parent asset name starts with ‘A’')
+    for c in asset_name:
+        if c not in B26_DIGITS:
+            raise exceptions.AssetNameError('parent asset name contains invalid character:', c)
+    return True
+
+def compact_subasset_longname(string):
+    name_int = 0
+    for i, c in enumerate(string[::-1]):
+        name_int += (68 ** i) * SUBASSET_REVERSE[c]
+    return name_int.to_bytes((name_int.bit_length() + 7) // 8, byteorder='big')
+
+def expand_subasset_longname(raw_bytes):
+    integer = int.from_bytes(raw_bytes, byteorder='big')
+    if integer == 0:
+        return ''
+    ret = ''
+    while integer != 0:
+        ret = SUBASSET_DIGITS[integer % 68 - 1] + ret
+        integer //= 68
+    return ret
+
+def generate_random_asset ():
+    return 'A' + str(random.randint(26**12 + 1, 2**64 - 1))
```

Comment:
It would be an inconvenience (the asset issuance would be invalid when it confirms).

The odds of a collision are roughly 1 in 100 million if there are 600k numeric assets in existence.  I chose not to worry about it.